numpy/lib/function_base.py

```diff
@@ -1557,28 +1557,55 @@ def sort_complex(a):
         return b
 
 
-def _trim_zeros(filt, trim=None):
+def _trim_zeros(
+    filt,
+    trim=None,
+    axis=None,
+    *,
+    atol=None,
+    rtol=None,
+    return_lengths=None
+):
     return (filt,)
 
 
 @array_function_dispatch(_trim_zeros)
-def trim_zeros(filt, trim='fb'):
-    """
-    Trim the leading and/or trailing zeros from a 1-D array or sequence.
+def trim_zeros(
+    filt,
+    trim='fb',
+    axis=-1,
+    *,
+    atol=0,
+    rtol=0,
+    return_lengths=False
+):
+    """Remove values along a dimension which are zero along all other.
 
     Parameters
     ----------
-    filt : 1-D array or sequence
+    filt : array_like
         Input array.
     trim : str, optional
         A string with 'f' representing trim from front and 'b' to trim from
-        back. Default is 'fb', trim zeros from both front and back of the
-        array.
+        back. By default, zeros are trimmed from the front and back.
+    axis : int or sequence, optional
+        The axis or a sequence of axes to trim. If None all axes are trimmed.
+    atol : float, optional
+        Absolute tolerance with which a value is considered for trimming.
+    rtol : float, optional
+        Relative tolerance with which a value is considered for trimming.
+    return_lengths : bool, optional
+        Additionally return the number of trimmed samples in each dimension at
+        the front and back.
```

Comment:
I don't like functions that modify their number of returned values with parameters. It seems like a good idea to provide the user with the trim lengths but in the long run I'd like to see a depreciation of this parameter and that `trim_zeros` always returns trim lengths.

Reply:
It's a lousy name, but `arg_trim_zeros` might be a better name for the function returning indices.

Reply:
Not enthusiastic about both options. Is there a way to deprecate the number of returned values? E.g. start with default value `False` and warn when `return_lengths=False` is given. Then after a deprecation cycle that parameter could be removed. If you don't think its worth going through a deprecation for this I'd prefer keeping this parameter over adding another utility function.

In case we'd add a separate function `arg_trim_zeros` I would agree that returning the actual indices would be more useful. However I think the current behavior is more fitting because I think the primary use case of returning the indices (slicing) is already taken care of inside the function. I imagine this argument to be useful for cases where a user is interested in the size of the shared zero area. E.g. my original reason for implementing this was to reduce the size of a bunch of nd-signals while still being able to reconstruct the original signal by padding. I see this as an analogue to the `pad_width` parameter of `numpy.pad`.

Reply:
> because I think the primary use case of returning the indices (slicing) is already taken care of inside the function

What about the case where I want to trim one array based on the non-zero area of another?